java-squid/src/test/java/org/sonar/java/se/symbolicvalues/SymbolicValueRelationsTest.java

```diff
@@ -0,0 +1,619 @@
+    checkTransitive(LessThanOrEqualRelation.class, MethodEqualsRelation.class, LessThanOrEqualRelation.class, RelationState.FULFILLED);
+    checkTransitive(LessThanOrEqualRelation.class, NotMethodEqualsRelation.class, LessThanOrEqualRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanOrEqualRelation.class, GreaterThanOrEqualRelation.class, LessThanOrEqualRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanOrEqualRelation.class, GreaterThanRelation.class, LessThanOrEqualRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanOrEqualRelation.class, LessThanOrEqualRelation.class, LessThanOrEqualRelation.class, RelationState.FULFILLED);
+    checkTransitive(LessThanOrEqualRelation.class, LessThanRelation.class, LessThanOrEqualRelation.class, RelationState.FULFILLED);
+  }
+
+  @Test
+  public void transitiveLessThan() {
+    checkTransitive(LessThanRelation.class, EqualRelation.class, LessThanRelation.class, RelationState.FULFILLED);
+    checkTransitive(LessThanRelation.class, NotEqualRelation.class, LessThanRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanRelation.class, MethodEqualsRelation.class, LessThanRelation.class, RelationState.FULFILLED);
+    checkTransitive(LessThanRelation.class, NotMethodEqualsRelation.class, LessThanRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanRelation.class, GreaterThanOrEqualRelation.class, LessThanRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanRelation.class, GreaterThanRelation.class, LessThanRelation.class, RelationState.UNDETERMINED);
+    checkTransitive(LessThanRelation.class, LessThanOrEqualRelation.class, LessThanRelation.class, RelationState.FULFILLED);
+    checkTransitive(LessThanRelation.class, LessThanRelation.class, LessThanRelation.class, RelationState.FULFILLED);
+  }
+
+  @Test
+  public void checkEquality() {
+    SymbolicValue a = values[0];
+    SymbolicValue b = values[1];
+    SymbolicValue c = values[2];
+    SymbolicValueRelation rel = new EqualRelation(a, b);
+    assertThat(rel).isNotEqualTo(new String());
+    assertThat(rel).isEqualTo(new EqualRelation(a, b));
+    assertThat(rel).isNotEqualTo(new NotEqualRelation(a, b));
+    assertThat(rel).isNotEqualTo(new EqualRelation(a, c));
+  }
+
+  @Test
+  public void checkTransitiveLimit() {
+    List<SymbolicValueRelation> relations = new ArrayList<>();
+    SymbolicValue first = null;
+    SymbolicValue previous = null;
+    SymbolicValue last = null;
+    for (int i = 0; i < 300; i++) {
+      last = new SymbolicValue(i);
+      if (first == null) {
+        first = last;
+      } else if (previous != null) {
+        relations.add(new EqualRelation(previous, last));
+      }
+      previous = last;
+    }
+    try {
+      new EqualRelation(first, last).resolveState(relations);
```

Comment:
This test will not fail is the limit is not reach ! this is an assertion less test ! 
it should fail (so have a fail assertion) if the exception is not raised!! Moreover, I would expect to have an assertion on the excpetion raised to be sure that we raise it for the correct reasons.